comp/metadata/inventoryotel/inventoryotelimpl/inventoryotel.go

```diff
@@ -0,0 +1,270 @@
+	"github.com/DataDog/datadog-agent/comp/api/api/utils"
+	"github.com/DataDog/datadog-agent/comp/api/authtoken"
+	"github.com/DataDog/datadog-agent/comp/core/config"
+	flaretypes "github.com/DataDog/datadog-agent/comp/core/flare/types"
+	"github.com/DataDog/datadog-agent/comp/core/log"
+	"github.com/DataDog/datadog-agent/comp/core/status"
+	"github.com/DataDog/datadog-agent/comp/metadata/internal/util"
+	iointerface "github.com/DataDog/datadog-agent/comp/metadata/inventoryotel"
+	"github.com/DataDog/datadog-agent/comp/metadata/runner/runnerimpl"
+	"github.com/DataDog/datadog-agent/pkg/serializer"
+	"github.com/DataDog/datadog-agent/pkg/serializer/marshaler"
+	"github.com/DataDog/datadog-agent/pkg/util/fxutil"
+	"github.com/DataDog/datadog-agent/pkg/util/hostname"
+	"github.com/DataDog/datadog-agent/pkg/util/scrubber"
+	"github.com/DataDog/datadog-agent/pkg/util/uuid"
+)
+
+// Module defines the fx options for this component.
+func Module() fxutil.Module {
+	return fxutil.Component(
+		fx.Provide(newInventoryOtelProvider))
+}
+
+type otelMetadata = map[string]interface{}
+
+// Payload handles the JSON unmarshalling of the metadata payload
+type Payload struct {
+	Hostname  string       `json:"hostname"`
+	Timestamp int64        `json:"timestamp"`
+	Metadata  otelMetadata `json:"otel_metadata"`
+	UUID      string       `json:"uuid"`
+}
+
+// MarshalJSON serialization a Payload to JSON
+func (p *Payload) MarshalJSON() ([]byte, error) {
+	type PayloadAlias Payload
+	return json.Marshal((*PayloadAlias)(p))
+}
+
+// SplitPayload implements marshaler.AbstractMarshaler#SplitPayload.
+//
+// In this case, the payload can't be split any further.
+func (p *Payload) SplitPayload(_ int) ([]marshaler.AbstractMarshaler, error) {
+	return nil, fmt.Errorf("could not split inventories agent payload any more, payload is too big for intake")
+}
+
+type inventoryotel struct {
+	util.InventoryPayload
+
+	conf      config.Component
+	log       log.Component
+	m         sync.Mutex
+	data      otelMetadata
+	hostname  string
+	authToken authtoken.Component
+	f         *freshConfig
+}
+
+type dependencies struct {
+	fx.In
+
+	Log        log.Component
+	Config     config.Component
+	Serializer serializer.MetricSerializer
+	AuthToken  authtoken.Component
+}
+
+type provides struct {
+	fx.Out
+
+	Comp                 iointerface.Component
+	Provider             runnerimpl.Provider
+	FlareProvider        flaretypes.Provider
+	StatusHeaderProvider status.HeaderInformationProvider
+	Endpoint             api.AgentEndpointProvider
+}
+
+func newInventoryOtelProvider(deps dependencies) provides {
+	hname, _ := hostname.Get(context.Background())
+	i := &inventoryotel{
+		conf:      deps.Config,
+		log:       deps.Log,
+		hostname:  hname,
+		data:      make(otelMetadata),
+		authToken: deps.AuthToken,
+	}
+
+	getter := i.fetchRemoteOtelConfig
+	if i.conf.GetBool("otel.submit_dummy_inventories_metadata") {
+		getter = i.fetchDummyOtelConfig
```

Comment:
What is the use case for this ?

Reply:
This will be removed; but we currently need it while in development to send a payload we can use to develop the backend services, REDAPL, etc... while the datadog extension in the `otel-agent` is in the works. This will be killed prior to merging into `main` I would reckon, or shortly after. Honestly, this PR could be a draft PR until the extension is ready.